server-main.go

```diff
@@ -137,6 +137,15 @@ func initServerConfig(c *cli.Context) {
 	accessKey := os.Getenv("MINIO_ACCESS_KEY")
 	secretKey := os.Getenv("MINIO_SECRET_KEY")
 
+	// Fetch max conn limit from environment variable.
+	maxConnStr := os.Getenv("MINIO_MAXCONN")
+	if maxConnStr != "" {
+		// We need to parse to its integer value.
+		maxConn, err := strconv.Atoi(maxConnStr)
+		fatalIf(err, "Unable to convert MINIO_MAXCONN=%s environment variable into its integer value.", maxConnStr)
+		globalMaxConn = maxConn
+	}
+
```

Comment:
// Enable connection limit if MINIO_MAXCONN is set.
if maxConnStr := os.Getenv("MINIO_MAXCONN");  maxConnStr != "" {
		globalMaxConn, err := strconv.Atoi(maxConnStr)
		fatalIf(err, "Unable to convert MINIO_MAXCONN=%s environment variable into its integer value.", maxConnStr)
}